Pear/Cells/EditDemographicsCollectionViewCell.swift

```diff
@@ -0,0 +1,342 @@
+            if tag == 2 {
+                demographics.major = valueSelected
+            }
+            if tag == 3 {
+                demographics.hometown = valueSelected
+            }
+            if tag == 4 {
+                demographics.pronouns = valueSelected
+            }
+        }
+    }
+
+    /// Resizes search view based on input to search field.
+    func updateDropdownViewHeight(dropdownView: UIView, height: CGFloat) {
+        editScrollView.bringSubviewToFront(dropdownView)
+        updateFieldConstraints(fieldView: dropdownView, height: textFieldHeight + height)
+    }
+
+    /// Brings field view to the front of the screen and handles keyboard interactions
+    /// when switching from select dropdowns to search.
+    func bringDropdownViewToFront(dropdownView: UIView, height: CGFloat, isSelect: Bool) {
+        if let activeDropdownView = activeDropdownView as? OnboardingSearchDropdownView,
+            activeDropdownView != dropdownView {
+            editScrollView.sendSubviewToBack(activeDropdownView)
+            updateFieldConstraints(fieldView: activeDropdownView, height: textFieldHeight)
+            activeDropdownView.hideTableView()
+            activeDropdownView.endEditing(true)
+        } else if let activeDropdownView = activeDropdownView as? OnboardingSelectDropdownView,
+            activeDropdownView != dropdownView {
+            editScrollView.sendSubviewToBack(activeDropdownView)
+            updateFieldConstraints(fieldView: activeDropdownView, height: textFieldHeight)
+            activeDropdownView.hideTableView()
+        } else if let activeDropdownView = activeDropdownView as? UITextField {
+            activeDropdownView.endEditing(true)
+        }
+        editScrollView.bringSubviewToFront(dropdownView)
+        activeDropdownView = dropdownView
+        let newFieldHeight = textFieldHeight + height
+        updateFieldConstraints(fieldView: dropdownView, height: newFieldHeight)
+        // Scroll screen up if dropdown field extends past bottom of screen
+        if dropdownView.frame.origin.y + newFieldHeight > contentView.bounds.height - 50 {
+            isPageScrolled = true
+            editScrollView.setContentOffset(CGPoint(x: 0, y: height), animated: true)
+        }
+    }
+
+    /// Send field view to the back of the screen to allow interactions with other field views.
+    func sendDropdownViewToBack(dropdownView: UIView) {
+        contentView.endEditing(true)
+        editScrollView.sendSubviewToBack(dropdownView)
+        updateFieldConstraints(fieldView: dropdownView, height: textFieldHeight)
+        if isPageScrolled {
+            editScrollView.setContentOffset(CGPoint(x: 0, y: 0), animated: true)
+            isPageScrolled = false
+        }
+    }
+}
+
+extension EditDemographicsCollectionViewCell: UIImagePickerControllerDelegate,
+                                          UINavigationControllerDelegate,
+                                          UITextFieldDelegate {
```

Comment:
I don't like this 😂
No clue if this is convention or not.
They should at least be aligned.